jolt-sdk/macros/src/lib.rs

```diff
@@ -11,19 +11,42 @@ use common::{
     },
     rv_trace::MemoryLayout,
 };
+use lazy_static::lazy_static;
 use proc_macro::TokenStream;
 use proc_macro2::TokenStream as TokenStream2;
 use quote::quote;
+use std::sync::Mutex;
 use syn::{
     parse_macro_input, AttributeArgs, Ident, ItemFn, Lit, Meta, MetaNameValue, NestedMeta, PatType,
     ReturnType, Type,
 };
 
+lazy_static! {
+    static ref WASM_IMPORTS_ADDED: Mutex<bool> = Mutex::new(false);
```

Comment:
Why are we using a lazy static here? We have many other instances where we dynamically generate code based on an attribute in the provable macro and we don't need to use these.

Reply:
I thought we could use lazy_static here because the macro code for WASM needs certain global imports and a helper structure for serialization to be initialized only once, regardless of how many functions are labelled with the `wasm` attribute..

Do you think the global initialization is a problem? If lazy_static is the issue, we could use Once from the std-library like this:
```
static WASM_IMPORTS_INIT: Once = Once::new();

#[proc_macro_attribute]
pub fn provable(attr: TokenStream, item: TokenStream) -> TokenStream {
    let attr = parse_macro_input!(attr as AttributeArgs);
    let func = parse_macro_input!(item as ItemFn);
    let mut builder = MacroBuilder::new(attr, func);

    let mut token_stream = builder.build();

    // Add wasm utilities and functions if the function is marked as wasm
    if builder.has_wasm_attr() {
        // wasm utilities should only be added once
        WASM_IMPORTS_INIT.call_once(|| {
            let wasm_utilities: TokenStream = builder.make_wasm_utilities().into();
            token_stream.extend(wasm_utilities);
        });
        let wasm_token_stream: TokenStream = builder.make_wasm_function().into();
        token_stream.extend(wasm_token_stream);
    }

    token_stream
}
```

Would this approach be more suitable for our use case?

Reply:
Oh I see what you're doing here now. I think this is fine.